ffi/diplomat/src/provider.rs

```diff
@@ -23,7 +24,7 @@ pub enum ICU4XDataProviderInner {
 
 impl Default for ICU4XDataProviderInner {
     fn default() -> Self {
-        Self::Empty
+        Self::Destroyed
     }
 }
```

Comment:
Suggestion: Your change is making there be an ambiguous default. It would be more explicit if we didn't have a Default impl and instead used core::mem::replace instead of core::mem::take